### PR TITLE
[SC-6819] Allowance Flow in Xstate for AAVE open & manage

### PR DESCRIPTION
**Pull request by jakubswierczek**
# [Allowance ❤️ Xstate](https://app.shortcut.com/oazo-apps/story/6818/add-allowance-flow-to-xstate-architecture)

## Changes 👷‍♀️
- Allowance State Machine with Xstate
- If `token` for manage or open is different than `ETH` the flow requires allowance steps
- 
## How to test 🧪
- change token in strategy config to not `ETH`
- enjoy the allowance step.

### CODE REVIEW

File: features/aave/open/state/openAaveStateMachine.stories.tsx [hunk 1]
```diff
@@ -24,13 +24,16 @@ const hasGasEstimation: HasGasEstimation = {
   gasEstimation: 10,
 }
 
-const openAaveStateMachine = createOpenAaveStateMachine({} as any, {} as any, {} as any).withConfig(
-  {
-    services: {
-      ...({} as OpenAaveStateMachineServices),
-    },
+const openAaveStateMachine = createOpenAaveStateMachine(
+  {} as any,
```
Comment (marcinciarka) [inline features/aave/open/state/openAaveStateMachine.stories.tsx:28]: We could move `{} as any` to a helper and reuse it

File: features/aave/manage/services/getManageAavePositionStateMachineServices.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ export function getManageAavePositionStateMachineServices(
   prices$: (tokens: string[]) => Observable<Tickers>,
   strategyInfo$: (collateralToken: string) => Observable<IStrategyInfo>,
   aaveProtocolData$: (collateralToken: string, address: string) => Observable<AaveProtocolData>,
+  tokenAllowance$: (token: string, spender: string) => Observable<BigNumber>,
```
[the remark this answers is not in the document]
Comment (jakubswierczek) [reply features/aave/manage/services/getManageAavePositionStateMachineServices.ts:26]:
```
interface TokenAllowanceArgs {
  token: string
  owner: string
  spender: string
}
```
`tokenAllowance` has 3 parameters. I created a wrapper for only a token and a spender. In that case, we only want to check the allowance when the user has a proxy.

**Review by marcinciarka — COMMENTED** [review]
image
There seems to be no default value (amount) of token that's not in my wallet (changed to DAI while I only had ETH).

**Review by marcinciarka — APPROVED** [review]
:v:

**Pull request merged**